builtin/mainmenu/dlg_rebind_keys.lua

```diff
@@ -0,0 +1,101 @@
+-- Luanti
+-- SPDX-License-Identifier: LGPL-2.1-or-later
+-- Modified based on dlg_reinstall_mtg.lua
+-- Note that this is only needed for migrating from <5.11 to 5.12.
+
+local doc_url = "https://docs.luanti.org/for-players/controls/"
+
+local function get_formspec(dialogdata)
+	local markup = table.concat({
+		"<big>" .. hgettext("Keybindings changed") .. "</big>",
+		hgettext("The internal input handling system was reworked in Luanti 5.12.0."),
+		hgettext("As a result, your keybindings may have been changed."),
+		hgettext("Refer to the following website for more information:"),
+		("<action name='doc_url'><style color='cyan' hovercolor='orangered'>%s</style></action>"):format(doc_url),
+	}, "\n")
+
+	return table.concat({
+		"formspec_version[6]",
+		"size[12,7]",
+		"hypertext[0.5,0.5;11,4.7;text;", core.formspec_escape(markup), "]",
+		"container[0.5,5.7]",
+		"style[dismiss;bgcolor=red]",
+		"button[0,0;4,0.8;dismiss;", fgettext("Dismiss"), "]",
+		"button[4.5,0;6.5,0.8;reconfigure;", fgettext("Configure keybindings"), "]",
```

Comment:
(in combination with other suggestion)
```suggestion
		"button[4.5,0;6.5,0.8;reconfigure;", fgettext("Open settings"), "]",
```